### PR TITLE
feat: Handles case where we overflow void elements.

### PR DESCRIPTION
**Pull request by damencho**
Handles more tags and tracks.

### CODE REVIEW

**Comment by damencho** [comment]
We heavily depend on the RandomAccessFile from FileDataWriter.

So I suggest dropping DataWriter, always using FileDataWriter, and getting rid of isSeekable. 

@bgrozev WDYT?

**Comment by damencho** [comment]
Well there is some logic behind isSeekable() ... my current changes may break that ... but works for our case though and I have added some tests that cover this.

**Comment by damencho** [comment]
Anway even without isSeekable() those usecases are broken tests that were added prove that.

**Comment by damencho** [comment]
Looking at the code that I removed:
```
long len = tagsElem.writeElement(ioDW);
    if (ioDW.isSeekable())
    {
      new VoidElement(BLOCK_SIZE - tagsElem.getTotalSize()).writeElement(ioDW);
      return BLOCK_SIZE;
    }
```
Seems that the logic of isSeekable is that if it is not seekable you do not add void element, as you will not be able to seek and come back and fill it. 
And the MatroskaFileWriter.close is processing the file only when isSeekable is true, so no problem there.

File: src/test/java/org/ebml/matroska/MatroskaFileWriterTest.java [hunk 1]
```diff
@@ -117,6 +117,65 @@ public void testMultipleTracks() throws Exception
...
 
+  @Test
+  public void testMultipleTracksToFillVoidArea() throws Exception
+  {
+    final MatroskaFileWriter writer = new MatroskaFileWriter(ioDW);
+
+    for (int i = 0; i < 40; i++)
+    {
+      final MatroskaFileTrack nextTrack = new MatroskaFileTrack();
+      nextTrack.setTrackNo(i + 2);
```
Comment (bgrozev) [inline src/test/java/org/ebml/matroska/MatroskaFileWriterTest.java]: The `+2` is only necessary to make line 135 work? Can we simplify?

**Review by bgrozev — APPROVED** [review]

File: src/test/java/org/ebml/matroska/MatroskaFileWriterTest.java (hunk 1 above)
Comment (damencho) [reply src/test/java/org/ebml/matroska/MatroskaFileWriterTest.java]: Actually no idea why it is there. I dropped it.

**Pull request merged**